src/test_utils/kbn_server.js

```diff
@@ -25,7 +25,10 @@ import KbnServer from '../../src/server/kbn_server';
 
 const DEFAULTS_SETTINGS = {
   server: {
-    autoListen: false,
+    autoListen: true,
```

Comment:
__note:__ I have mixed feelings on that, but can't think of any other better way :/ We should start core's root to have access to the underlying server and it's only started when legacy hapi servers calls `listen`.

Even after https://github.com/elastic/kibana/pull/19994, we won't register routes and create Hapi server if `autoListen` is false, and we don't want to expose Hapi and hence `inject` either. So we need to think of a better alternative to `testKbnServer` at some point.

Reply:
Hmm, is there value in splitting up the logic in the new platform like it's split in the legacy platform, so the server is setup (routes registered, etc.) without listening, and then when calling `listen()` on the legacy platform would somehow tell the new platform to start listening too?

Reply:
> Hmm, is there value in splitting up the logic in the new platform like it's split in the legacy platform, so the server is setup (routes registered, etc.) without listening, and then when calling `listen()` on the legacy platform would somehow tell the new platform to start listening too?

Maybe, we can discuss that when we come closer to the plugin support when we start thinking where in the server lifecycle we do route registration. IIRC one of the reasons of putting this kind of staff in `start/stop` was based on the idea that someday we may want to stop service/server, reload config (not only logging, but the one that can affect routes, http config etc.) and start service/server again. 

> The primary reason I ask is that given the new platforms dependency on listen for setup, it seems that there is little point in having a non-listening KbnServer anymore... 

I think it's still needed in dev mode when `kbnServer` is created by `optimizer` worker process or what you mean?

Reply:
> I think it's still needed in dev mode when kbnServer is created by optimizer worker process or what you mean?

But the optimizer starts a listening version of the Kibana server, which is proxied to by the actual server, so it can serve the optimizer output. 


What I mean is that even if the `KbnServer` consumer doesn't need HTTP to be bound for its purposes, the `KbnServer` must bind to a port to function properly, so I feel like we should get rid of the `listen()` method and just make it a part of `KbnServer` initialization, part of the promise that is returned by `ready()`, and remove the `server.autoListen` config option.

Reply:
Whoops, I'm totally wrong... see https://github.com/elastic/kibana/blob/4ebd3d353409e7bff681d1aa35e8d0087e1af121/src/cli/cluster/cluster_manager.js#L51

Ignore my request to remove `server.autoListen` :)

We pass `--server.autoListen=false` to the optimizer, which writes the bundle assets to the bundle directory and the actual server just pauses requests while the optimizer is running, then serves the requests from the bundle directory... This means that the server is semi-functional, it just can't handle routing via the `kbnServer.inject()` method, which won't be a problem as we move further toward the new platform.